include/api/wasmedge/wasmedge.h

```diff
@@ -2636,6 +2658,13 @@ WASMEDGE_CAPI_EXPORT extern WasmEdge_MemoryInstanceContext *
 WasmEdge_CallingFrameGetMemoryInstance(const WasmEdge_CallingFrameContext *Cxt,
                                        const uint32_t Idx);
 
+/// Suspend the execution from the current calling frame.
+/// This is only available when using WasmEdge_Coroutine.
+///
+/// \param Cxt the WasmEdge_CallingFrameContext.
+WASMEDGE_CAPI_EXPORT extern void
+WasmEdge_CallingFrameCoroutineSuspend(const WasmEdge_CallingFrameContext *Cxt);
```

Comment:
How is this expected to be used? Is another thread expected to call this?

Ideally for our use case we would suspend execution periodically using gas metering

Reply:
Hi @rockwotj 
This feature is for those who want to suspend the execution from the host functions. We have some use cases that will spend much time waiting for IO. They would like to suspend the execution until the IO is ready and get it back.

Reply:
I see. We do this today using `std::longjmp` to jump out from the host function then back in on a separate stack.

An important use case for our thread-per-core architecture is that one task doesn't hog the CPU for too long. Otherwise that means other requests that thread is responsible for cannot be responded, increasing tail latencies in these situations. See https://github.com/scylladb/seastar/blob/master/doc/tutorial.md#breaking-up-long-running-computations

Anyways, it seems this mechanism and set of APIs won't solve this use case and instead we need some callback function to be periodically invoked from the runtime while executing.